agenta-web/src/components/EvaluationTable/ABTestingEvaluationTable.tsx

```diff
@@ -96,13 +97,43 @@ const ABTestingEvaluationTable: React.FC<EvaluationTableProps> = ({
     const [rows, setRows] = useState<ABTestingEvaluationTableRow[]>([])
     const [evaluationStatus, setEvaluationStatus] = useState<EvaluationFlow>(evaluation.status)
     const [evaluationResults, setEvaluationResults] = useState<any>(null)
+    const [rowData, setRowData] = useState<
+        {appVariant1: string; inputs: string; vote: string; appVariant0: string}[]
+    >([])
+    const [columnDefs, setColumnDefs] = useState<{field: string; [key: string]: any}[]>([])
 
     useEffect(() => {
         if (evaluationScenarios) {
             setRows(evaluationScenarios)
         }
     }, [evaluationScenarios])
 
+    useEffect(() => {
+        const getRows = rows.map((data) => {
+            return {
+                inputs: data.inputs[0].input_value,
+                appVariant0: data?.columnData0
+                    ? data?.columnData0
+                    : data.outputs[0]?.variant_output,
+                appVariant1: data?.columnData1
+                    ? data?.columnData1
+                    : data.outputs[1]?.variant_output,
+                vote: data.vote,
+            }
+        })
+        setRowData(getRows)
+    }, [rows])
+
+    useEffect(() => {
+        if (Array.isArray(rowData) && rowData.length > 0 && typeof rowData[0] === "object") {
```

Comment:
No need to check `typeof rowData[0] === "object"`.

`Array.isArray` is sufficient